python/chainer_compiler.py

```diff
@@ -162,16 +168,52 @@ def backward(self, indexes, flat_gys):
         return gxs
 
 
+@contextmanager
+def enable_unified_memory(use):
+    if use:
+        cupy.cuda.set_allocator(cupy.cuda.memory.malloc_managed)
+    try:
+        yield None
+    finally:
+        if use:
+            memory_pool = cupy.get_default_memory_pool()
+            cupy.cuda.set_allocator(memory_pool.malloc)
+
+
+def _run_translator(translator, mc, inputs):
+    if translator == 'ch2o':
+        xmodel = ch2o.compile_model(mc, inputs)
+        f = tempfile.NamedTemporaryFile(delete=False)
+        f.write(xmodel.SerializeToString())
+        f.close()
+        del xmodel
+    elif translator == 'onnx_chainer':
+        import onnx_chainer
+        f = tempfile.NamedTemporaryFile(delete=False)
+        onnx_chainer.export(mc, inputs, filename=f)
+        f.close()
+    else:
+        raise NotImplementedError('Unsupported translator:',
+                                  translator)
+
+    graph = chainer_compiler_core.load(f.name)
+    os.unlink(f.name)
+
+    return graph
+
+
 class CompiledModel(chainer.Chain):
 
     def __init__(self, model, inputs, translator='ch2o', dump_onnx=False,
-                 computation_order=None):
+                 computation_order=None,
+                 use_unified_memory=False):
```

Comment:
As chatted, it would be better to change the name of this argument to `allocators` or something which will be a tuple of (runtime allocator, export time allocator).

Reply:
I added arguments of export_allocator and runtime_allocator.